### PR TITLE
Remove latestDependencies variant

### PR DESCRIPTION
**Pull request by vegaro**
I noticed we still had the `latestDependencies` flavor so I removed it

### CODE REVIEW

**Review by NachoSoto — APPROVED** [review]

File: android/gradle.properties [hunk 1]
```diff
@@ -41,6 +41,5 @@ POM_DEVELOPER_ID=revenuecat
 POM_DEVELOPER_NAME=RevenueCat, Inc.
 #Do not sign releases. When calling uploadArchives pass -PRELEASE_SIGNING_ENABLED=true
 RELEASE_SIGNING_ENABLED=false
-ANDROID_VARIANT_TO_PUBLISH=latestDependenciesRelease
```
Comment (vegaro) [inline android/gradle.properties:44]: this is `release` by default

Comment (tonidero) [reply android/gradle.properties:44]: I know gradle added support for deploying multiple variants. We should make sure after removing this we aren't also deploying the debug variant.

**Review by tonidero — APPROVED** [review]

File: android/gradle.properties (hunk 1 above)
Comment (vegaro) [reply android/gradle.properties:44]: Yes, I checked publishing to maven local and this is what's in the .module so only that variant is being published

```
{
  "formatVersion": "1.1",
  "component": {
    "group": "com.revenuecat.purchases",
    "module": "purchases-hybrid-common",
    "version": "5.2.2",
    "attributes": {
      "org.gradle.status": "release"
    }
  },
  "createdBy": {
    "gradle": {
      "version": "7.5"
    }
  },
  "variants": [
    {
      "name": "releaseVariantReleaseApiPublication",
      "attributes": {
        "org.gradle.category": "library",
        "org.gradle.dependency.bundling": "external",
        "org.gradle.libraryelements": "aar",
        "org.gradle.usage": "java-api"
      },
      "dependencies": [
        ...
      ],
      "files": [
        {
          "name": "purchases-hybrid-common-5.2.2.aar",
          "url": "purchases-hybrid-common-5.2.2.aar",
          "size": 71369,
          ...
        }
      ]
    },
    {
      "name": "releaseVariantReleaseRuntimePublication",
      "attributes": {
        "org.gradle.category": "library",
        "org.gradle.dependency.bundling": "external",
        "org.gradle.libraryelements": "aar",
        "org.gradle.usage": "java-runtime"
      },
      "dependencies": [
        ...
      ],
      "files": [
        {
          "name": "purchases-hybrid-common-5.2.2.aar",
          "url": "purchases-hybrid-common-5.2.2.aar",
          "size": 71369,
          ...
        }
      ]
    },
    {
      "name": "releaseVariantReleaseSourcePublication",
      "attributes": {
        "org.gradle.category": "documentation",
        "org.gradle.dependency.bundling": "external",
        "org.gradle.docstype": "sources",
        "org.gradle.usage": "java-runtime"
      },
      "files": [
        {
          "name": "purchases-hybrid-common-5.2.2-sources.jar",
          "url": "purchases-hybrid-common-5.2.2-sources.jar",
          "size": 15818,
          ...
        }
      ]
    },
    {
      "name": "releaseVariantReleaseJavaDocPublication",
      "attributes": {
        "org.gradle.category": "documentation",
        "org.gradle.dependency.bundling": "external",
        "org.gradle.docstype": "javadoc",
        "org.gradle.usage": "java-runtime"
      },
      "files": [
        {
          "name": "purchases-hybrid-common-5.2.2-javadoc.jar",
          "url": "purchases-hybrid-common-5.2.2-javadoc.jar",
          "size": 336706,
          ...
        }
      ]
    }
  ]
}
```

I believe we would need to use the base plugin of maven-gradle-plugin to configure multi variant. Take a look at this changelog https://vanniktech.github.io/gradle-maven-publish-plugin/changelog/#0210-2022-07-11

Comment (tonidero) [reply android/gradle.properties:44]: Makes sense, thanks for checking!

**Pull request merged**